pytest/tests/sanity/replay_chain_from_archive.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+# Spins up 4 validating nodes, 1 non-validating RPC node, and 1 non-validating archival node.
+# There are four shards in this test.
+# Issues a number of transactions on the chain, and then replays the chain from the archival node.
+
+import copy
+import pathlib
+import subprocess
+import sys
+import unittest
+
+sys.path.append(str(pathlib.Path(__file__).resolve().parents[2] / 'lib'))
+
+from configured_logger import logger
+import cluster
+import simple_test
+import state_sync_lib
+
+EPOCH_LENGTH = 10
+GC_NUM_EPOCHS_TO_KEEP = 3
+GC_BLOCKS_LIMIT = 5
+
+# Shard layout with 5 roughly equal size shards for convenience.
+SHARD_LAYOUT = {
+    "V1": {
+        "boundary_accounts": [
+            "fff",
+            "kkk",
+            "ppp",
+            "uuu",
+        ],
+        "version": 2,
+        "shards_split_map": [],
+        "to_parent_shard_map": [],
+    }
+}
+
+NUM_SHARDS = len(SHARD_LAYOUT["V1"]["boundary_accounts"]) + 1
+
+ALL_ACCOUNTS_PREFIXES = [
+    "aaa",
+    "ggg",
+    "lll",
+    "rrr",
+    "vvv",
+]
+
+
+class ReplayChainFromArchiveTest(unittest.TestCase):
+
+    def setUp(self):
+        node_config_dump, node_config_sync = state_sync_lib.get_state_sync_configs_pair(
+        )
+
+        # Validator node configs: Enable single-shard tracking with memtries enabled.
+        node_config_sync["tracked_shards"] = []
+        configs = {x: node_config_sync for x in range(4)}
+
+        # Dump+RPC node config: Enable tracking all shards with memtries enabled.
+        node_config_dump["tracked_shards"] = [0]
+        configs[4] = node_config_dump
+
+        # Archival node config: Enable tracking all shards with memtries enabled.
+        node_config_archival = copy.deepcopy(node_config_dump)
+        node_config_archival["archive"] = True
```

Comment:
I suppose this archival node is using the split db configuration. Can you confirm it does?

Reply:
yes, confirmed in the directory and config files that cold db is indeed generated.

we have code in nayduck library to configure the cold db when `archive` is set to true:
https://github.com/near/nearcore/blob/2a44f185b32f7490f17441e3a8385ae9907d1c78/pytest/lib/cluster.py#L914